js/filters.js

```diff
@@ -177,4 +183,73 @@ TypeaheadFilter.prototype.handleNestedChange = function(e) {
   ]);
 };
 
+function ElectionFilter(elm) {
+  Filter.call(this, elm);
+
+  this.duration = parseInt(this.$body.data('duration'));
+  this.cycleName = this.$body.data('cycle-name');
+  this.fullName = this.$body.data('full-name');
+
+  this.$election = this.$body.find('.js-election');
+  this.$cycles = this.$body.find('.js-cycles');
+  this.$cycle = this.$body.find('input[type="hidden"][name="' + this.cycleName + '"]');
+  this.$full = this.$body.find('input[type="hidden"][name="' + this.fullName + '"]');
+
+  this.$election.on('change', this.handleElectionChange.bind(this));
+  this.$cycles.on('change', this.handleCycleChange.bind(this));
+
+  this.fields = [this.name, this.cycleName, this.fullName];
+}
+
+ElectionFilter.prototype = Object.create(Filter.prototype);
+ElectionFilter.constructor = ElectionFilter;
+
+ElectionFilter.prototype.fromQuery = function(query) {
+  var election = query[this.name] || '2016';
+  var cycle = query[this.cycleName] || election;
+  var full = query[this.fullName] !== null ? query[this.fullName] : true;
+  if (election) {
+    this.$election.val(election);
+    this.handleElectionChange({target: this.$election});
+    this.$cycles.find('input[value="' + cycle + ':' + full + '"]')
+      .prop('checked', true)
+      .change();
+  }
+  return this;
+};
+
+ElectionFilter.prototype.handleChange = function() {};
+
+ElectionFilter.prototype.handleElectionChange = function(e) {
+  if (this.duration === 2) {
+    return;
+  }
+  var election = parseInt($(e.target).val());
+  var cycles = _.range(election, election - this.duration, -2);
+  var bins = _.map(cycles, function(cycle) {
+    return {
+      name: this.name,
+      cycle: cycle,
+      min: cycle - 1,
+      max: cycle,
+      full: false
+    };
+  }.bind(this));
+  bins.unshift({
+    name: this.name,
+    cycle: election,
+    min: election - this.duration + 1,
+    max: election,
+    full: true
+  });
+  this.$cycles.html(cyclesTemplate(bins));
+  this.$cycles.find('input').eq(0).prop('selected', true).change();
```

Comment:
Small thing, but in other toggles we use the attribute "checked". Not sure if you care. Looks like it does the same thing.